added componentRestrictions paramater

pionl opened the pull request.

Commit:
added componentRestrictions paramater

flevour commented:
Hi @pionl , thanks for your contribution! Could you please change it so that `componentRestrictions` are a sub-key of `geocoderOptions`? This is to keep your change in line with the existing logic.

Commit:
moved componentRestrictions to geocoderOptions

pionl commented:
Hi @flevour , sure, I've missed that, much better. Fixed. Thanks

flevour commented on jquery.addressPicker.js:
```diff
@@ -64,7 +64,8 @@
                     appendAddressString: '',
-                    regionBias: ''
+                    regionBias: '',
+                	componentRestrictions : {}
```
@pionl last minor nitpick, could you fix indentation here please? :+1:

Commit:
fix indentation, added gitignore

pionl commented:
@flevour sure, changed to IDE, better? (the indentation wont appear in PHPStorm and sublime). I've also added .gitignore to prevent any unwanted system files or project based files. Is it ok? 
best regards,

flevour commented:
I'd prefer to leave the `.gitignore`. That kind ignores are better fitted into a global gitignore (see https://help.github.com/articles/ignoring-files/#create-a-global-gitignore or https://robots.thoughtbot.com/global-gitignore)

Thanks :) :roller_coaster: :rocket:

Commit:
removed gitignore

pionl commented:
Ok, sorry for the troubles.

flevour commented:
Thanks @pionl for your contribution!
